vllm/v1/request.py

```diff
@@ -1,18 +1,24 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import enum
+import functools
 from typing import TYPE_CHECKING, Optional, Union
 
-from vllm.lora.request import LoRARequest
+from vllm.logger import init_logger
 from vllm.sampling_params import SamplingParams
 from vllm.v1.engine import (EngineCoreEvent, EngineCoreEventType,
                             EngineCoreRequest, FinishReason)
+from vllm.v1.struct_output.request import StructuredOutputRequest
```

Comment:
Maybe under `TYPE_CHECKING`?

Reply:
This is used in `from_engine_core_request`, so it can't be in `TYPE_CHECKING`